### PR TITLE
pkg/binlog: parse statusVars and get ansi-quotes mode

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #878 , enable DM to get ansi-quotes mode more automaticly, so in future we could remove ansi-quotes in settings

### What is changed and how it works?
only added some function. `GetAnsiQuotesMode` or `statusVarsToKV` could be used on `replication.QueryEvent.StatusVars` in future

`statusVarsToKV` parses `replication.QueryEvent.StatusVars`, note that the order of KV sequence is undetermined (at least unordered I noticed).

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/binlog/event/util.go [hunk 1]
```diff
@@ -269,3 +271,170 @@ func writeStringColumnValue(buf *bytes.Buffer, value interface{}) error {
...
+			vars[key] = value
+			continue
+		}
+
+		// get variable-length value of according key and save it in `value`
+		switch key {
+		// 1-byte length + <length> chars of the catalog + '0'-char
+		case QCatalog:
+			if err = appendLengthThenCharsToValue(); err != nil {
+				return generateError()
```
Comment (csuzhangxc) [inline pkg/binlog/event/util.go:391]: can we carry `err` in the returned error?

Comment (lance6716) [reply pkg/binlog/event/util.go]: changed in https://github.com/pingcap/dm/pull/882/commits/7178f200203c5a86b14a2ea8009a6543bf27e990

**Comment by csuzhangxc** [comment]
How about also cherrying pick this to _release-1.0_ now?

**Comment by lance6716** [comment]
> How about also cherrying pick this to _release-1.0_ now?

now or after PR merging?

**Comment by csuzhangxc** [comment]
after PR merged. I mean cherry-pick it even it's just a part of #878.

**Pull request merged**